File: src/main/java/org/shark/example/rabbitMq/RabbitMqListener.java
package org.shark.example.rabbitMq;

import com.google.gson.Gson;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.service.rabbitmq.pojo.MessageQueueDto;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class RabbitMqListener {
    @RabbitListener(queues = {"example"}, containerFactory = "exampleContainerFactory")
    public void receiveMessage(MessageQueueDto message) {
        log.info("message: {}", new Gson().toJson(message));
    }
}

File: src/main/kotlin/org/shark/example/service/rabbitmq/RabbitMqListener.kt
package org.shark.example.service.rabbitmq

import com.google.gson.Gson
import org.shark.example.service.rabbitmq.pojo.MessageQueueDto
import org.slf4j.LoggerFactory
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Component

@Component
class RabbitMqListener {
    private val log = LoggerFactory.getLogger(RabbitMqListener::class.java)

    @RabbitListener(queues = ["example"], containerFactory = "exampleContainerFactory")
    fun receiveMessage(message: MessageQueueDto) {
        log.info("message: {}", Gson().toJson(message))
    }
}
